### DISCUSSION TITLE
Use composer for name instead of `Various Artists` or `Unknown Artist`

### DISCUSSION THREAD

**Discussion by rpgdev** [RogueOneEcho/caesura#128] [Ideas]
I'm getting this output for this particular torrent ID. Seems like the composer label throws it off or something. Transcode folders are prefixed with Unknown artist which of course is not ideal for upload.

$ caesura verify 2549589 --verbosity trace
2025-01-14 00:10:25.717 TRCE ↩ Sending request GET /ajax.php?action=torrent&id=2549589
2025-01-14 00:10:26.551 TRCE ↩ Received response after 0.834
2025-01-14 00:10:26.574 TRCE ↩ Sending request GET /ajax.php?action=torrentgroup&id=72626
2025-01-14 00:10:26.907 TRCE ↩ Received response after 0.333
2025-01-14 00:10:26.939 DBUG * Verifying Unknown Artist - ACTUAL ALBUM NAME [ACTUAL YEAR]
2025-01-14 00:10:26.945 TRCE ↩ Found 27 flacs in: ./content/ACTUAL ARTIST NAME - ACTUAL ALBUM NAME - FLAC

Censoring ACTUAL NAMES as extra precaution.

**Comment by RogueOneEcho** [comment]
It looks like RED has no artists listed which I've never seen before. So as the artists is empty caesura defaults to 
 `Unknown Artist`.

The quickest fix would be to add all the artists to RED then caesura will just show `Various Artists`. 

- [ ] Longer term, I may change the default to `Various Artists` and log a warning to the user that no artists are listed.

**Reply by rpgdev** [reply]
I think most people would rather have the composer identified as the artist in the folder name, like the original torrent does. The tool could also add a flag to read the artist from tag? Or pass it in as an argument?

**Reply by RogueOneEcho** [reply]
That's a good point.

In that case it's similar to #86

**Reply by rpgdev** [reply]
I see that's solved, is it solved in a way that I can make use of in this situation? I saw the commit code but nothing jumped at me as a flag or something that I could use.

**Reply by RogueOneEcho** [reply]
No, I was just noting it as a reminder to myself to re-read that discussion when I get around to solving the issue you've raised.

**Reply by rpgdev** [reply]
Makes sense, just wanted to confirm. Thanks for considering the change!

**Reply by RogueOneEcho** [reply]
I've looked through a lot of examples of soundtracks and I agree that where there are multiple artists but only a single composer the composer is typically used for the folder name.

Examples: 38990, 1321843, 5210512

Where things get more complex is with classical works. For classical works it's typically confusing to use the composer as there are so many different versions of famous composer's works that labelling something as "Bach" is generally avoided (although I can't recall where I read this).

However I think it's reasonable that if there are multiple main artists (more than 2) but only one composer it's preferable to use the composer for the torrent name rather than "Various Artists".

Examples: 4812051, 938669, 5201010

I did come across this example which contradicts my logic: 4675185 here everyone names by the composer when I would suggest the two artists are more logical.

**Reply by RogueOneEcho** [reply]
Fixed in d361e76b

**Reply by rpgdev** [reply]
It's a tough problem as tagging tends to be. In the future an option might be to pass in the artist yourself.. that way people have some control over it. Though since the edge cases are relatively few it may not matter. Until someone comes in and reports it, that is :)

**Reply by RogueOneEcho** [reply]
Indeed I'm pretty reliant on users reporting issues in order to improve the tool and move close to an optimal solution for all cases. If there were a `--name` option to override the `Artist - Album (Year) (Edition)` naming then 
a) People may misuse it and name things according to their own logic rather than adhering to standards
b) Edge cases like yours wouldn't get reported and therefore the logic built in to `caesura` wouldn't improve

**Reply by rpgdev** [reply]
Point A is addressed by each community’s rules and standards, which should regulate the kinds of names allowed. I wouldn’t necessarily expect a tool designed for different indexers to enforce this, at least not to the extent of hampering its own functionality. Point B is solid—can’t argue with that!
